packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -77,6 +93,38 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         uint128 amount0Min; // minimum amount of token0 to get from closing Uni LP
         uint128 amount1Min; // minimum amount of token1 to get from closing Uni LP
     }
+    // struct for flashloanWMintDepositNft() params
+    struct FlashloanWMintDepositNftParams {
+        uint256 vaultId; // vault ID (could be zero)
+        uint256 wPowerPerpAmount; // wPowerPerp amount to mint
+        uint256 collateralToDeposit; // ETH collateral amount to deposit in vault (could be zero)
+        uint256 collateralToFlashloan; // ETH amount to flashloan and use for deposit into vault
+        uint256 collateralToLp; // ETH collateral amount to use for LPing (could be zero)
+        uint256 collateralToWithdraw; // ETH amount to withdraw from vault (if collateralToLp>0, this should be = collateralToLp+fee or = collateralToLp and sender include fee in msg.value)
```

Comment:
this could also be 50% of the collateralToLp, right?

Reply:
not just = collateralToLp

Reply:
yup